lib/aggregate.ex

```diff
@@ -1630,10 +1630,13 @@ defmodule AshSql.Aggregate do
     field =
       case kind do
         :count ->
-          if Map.get(aggregate, :uniq?) do
-            Ecto.Query.dynamic([row], count(^field, :distinct))
-          else
-            Ecto.Query.dynamic([row], count())
+          cond do
+            Map.get(aggregate, :uniq?) ->
+              Ecto.Query.dynamic([row], count(^field, :distinct))
+            Ash.Resource.Info.field(resource, Ash.Query.Ref.name(ref)).allow_nil? ->
```

Comment:
I'd suggest if `match?(%{attribute: %{allow_nil?: false}}, ref)`

then we do the logic to count without the field. Otherwise, we do the current behavior.